develop/dev-guide-proxysql-integration.md

```diff
@@ -1,682 +1,1115 @@
-    ```
+    - ProxySQL tries to match the rules one by one in the order of `rule_id` from smallest to largest.
+    - `^` symbol matches the beginning of a SQL statement and `$` matches the end.
 
-    You can run the following statements:
+    For more information on ProxSQL regular expression and pattern matching, see [query_processor_regex](https://proxysql.com/documentation/global-variables/mysql-variables/#mysql-query_processor_regex) in ProxySQL documentation.
 
-    - `SELECT` statement:
+    For a full list of parameters, see [mysql_query_rules](https://proxysql.com/documentation/main-runtime/#mysql_query_rules) in ProxySQL documentation.
 
-        ```sql
-        SELECT * FROM test.test;
+6. Verify the configuration and check whether the query rules work.
+
+    1. Log into ProxySQL MySQL Interface as the `root` user:
+
+        <SimpleTab groupId="os">
+
+        <div label="macOS" value="macOS">
+
+        ```bash
+        mysql -u root -h 127.0.0.1 -P 6034
         ```
 
-        The statement is expected to match rules with `rule_id` of `2` and forward the statement to the TiDB server `tidb-1` with `hostgroup` of `1`.
+        </div>
 
-    - `SELECT ... FOR UPDATE` statement:
+        <div label="CentOS" value="CentOS">
 
-        ```sql
-        SELECT * FROM test.test for UPDATE;
+        ```bash
+        mysql -u root -h 127.0.0.1 -P 6034
+        ```
+
+        </div>
+
+        <div label="Windows (Git Bash)" value="Windows">
+
+        ```bash
+        mysql -u root -h 127.0.0.1 -P 6034
         ```
 
-        The statement is expected to match rules with `rule_id` of `1` and forward the statement to the TiDB server `tidb-0` with `hostgroup` of `0`.
+        </div>
 
-    - Transaction:
+        </SimpleTab>
+
+    2. Execute the following SQL statements:
+
+        - Execute a `SELECT` statement:
+
+            ```sql
+            SELECT * FROM test.tidb_server;
+            ```
+
+            This statement will match rule_id `2` and forward the statement to the TiDB server on `hostgroup 1`.
+
+        - Execute a `SELECT ... FOR UPDATE` statement:
+
+            ```sql
+            SELECT * FROM test.tidb_server for UPDATE;
+            ```
+
+            This statement will match rule_id `1` and forward the statement to the TiDB server on `hostgroup 0`.
+
+        - Start a transaction:
+
+            ```sql
+            BEGIN;
+            INSERT INTO test.tidb_server (server_name) VALUES ('insert this and rollback later');
+            SELECT * FROM test.tidb_server;
+            ROLLBACK;
+            ```
+
+            In this transaction, the `BEGIN` statement will not match any rules. It uses the default hostgroup (`hostgroup 0` in this example). Because ProxySQL enables user transaction_persistent by default, which will execute all statements within the same transaction in the same hostgroup, the `INSERT` and `SELECT * FROM test.tidb_server;` statements will also be forwarded to the TiDB server `hostgroup 0`.
```

Comment:
```suggestion
            In this transaction, the `BEGIN` statement will not match any rules. It uses the default hostgroup (`hostgroup 0` in this example). Because ProxySQL enables user transaction_persistent by default, which will execute all statements within the same transaction in the same hostgroup, the `INSERT` and `SELECT * FROM test.tidb_server;` statements will also be forwarded to the TiDB cluster `hostgroup 0`.
```